lua/luasnip/nodes/functionNode.lua

```diff
@@ -9,16 +8,108 @@ local key_indexer = require("luasnip.nodes.key_indexer")
 local opt_args = require("luasnip.nodes.optional_arg")
 local snippet_string = require("luasnip.nodes.util.snippet_string")
 
-local function F(fn, args, opts)
-	opts = opts or {}
+---@alias LuaSnip.FunctionNode.Fn fun(args: (string[])[], parent: LuaSnip.Snippet | LuaSnip.SnippetNode, ...: table): string|string[]
+-- FIXME: how to document each param of the callback function?
```

Comment:
Yeah that's a good question :+1:
I think it's not possible to document free-standing functions like that with LuaCATS :/
For reference:
https://github.com/EmmyLuaLs/emmylua-analyzer-rust/issues/779
https://github.com/LuaLS/lua-language-server/issues/1456

I think the current way of documenting these callbacks in the dynamicNode/functionNode signature is... fine, but not very elegant :/